backend/analytics_server/mhq/service/ai/ai_analytics_service.py

```diff
@@ -73,28 +97,28 @@ def _fireworks_ai_fetch_completions(self, messages: List[Dict[str, str]]):
         api_url = "https://api.fireworks.ai/inference/v1/chat/completions"
         response = requests.post(api_url, headers=self._headers, json=payload)
 
-        if response.status_code != HTTPStatus.OK:
-            raise Exception(response.json())
-
-        return response.json()
-
-    def _fetch_completion(self, messages: List[Dict[str, str]]):
+        return self._handle_api_response(response)
 
+    def _fetch_completion(
+        self, messages: List[Dict[str, str]]
+    ) -> Union[Dict[str, str], Dict[str, Union[str, int]]]:
+        """
+        Fetches the completion using the appropriate AI provider based on the LLM.
+        """
         if self._ai_provider == AIProvider.FIREWORKS_AI:
-            return self._fireworks_ai_fetch_completions(messages)["choices"][0][
-                "message"
-            ]["content"]
+            return self._fireworks_ai_fetch_completions(messages)
 
         if self._ai_provider == AIProvider.OPEN_AI:
-            return self._open_ai_fetch_completion_open_ai(messages)["choices"][0][
-                "message"
-            ]["content"]
+            return self._open_ai_fetch_completion_open_ai(messages)
 
-        raise Exception(f"Invalid AI provider {self._ai_provider}")
+        return {
+            "status": "error",
+            "message": f"Invalid AI provider {self._ai_provider}",
+        }
 
     def get_dora_metrics_score(
         self, four_keys_data: Dict[str, float]
-    ) -> Dict[str, str]:
+    ) -> Union[Dict[str, str], Dict[str, Union[str, int]]]:
```

Comment:
although this serves our usecase, this return type feels like an overkill

Reply:
and super hard to decipher

Reply:
I will make it simpler TBH I don't like this either